sections/product-recommendations.liquid

```diff
@@ -56,9 +58,14 @@
             const html = document.createElement('div');
             html.innerHTML = text;
             const recommendations = html.querySelector('product-recommendations');
+
             if (recommendations && recommendations.innerHTML.trim().length) {
               this.innerHTML = recommendations.innerHTML;
             }
+
+            if (html.querySelectorAll('.grid__item').length > 0) {
+              document.querySelector('product-recommendations').classList.add('product-recommendations--loaded');
```

Comment:
In the context of this custom HTML element, `this` should be targeting the `product-recommendations` element

```suggestion
              this.classList.add('product-recommendations--loaded');
```